### PR TITLE
Added a little clarification - implicit invocation of toString() in string templates

### PR DESCRIPTION
**Pull request by pichuzhkin**
Added info how templates really works - implicit invocation of toString()

### CODE REVIEW

**Comment by nikitabobko** [comment]
@pichuzhkin I'm curious what is your motivation for adding this clarification? What benefit does this clarification add?

The documentation should explain things, but it shouldn't turn into a verbose wall of obvious text, otherwise it'd be too hard to read it. There should be a balance.

Have you stumbled across some puzzler? Or the motivation is different?

File: docs/topics/strings.md [hunk 1]
```diff
@@ -111,14 +111,18 @@ By default, a pipe symbol `|` is used as margin prefix, but you can choose anoth
...
 
 ```kotlin
 fun main() {
 //sampleStart
     val i = 10
     println("i = $i") 
     // i = 10
+    val letters = listOf("a","b","c","d","e")
+    println("List of letters: $letters") // evaluates as letters.toString() 
+    // List of letters: [a, b, c, d, e]
```
Comment (nikitabobko) [inline docs/topics/strings.md:125]: It's better to mention something with an explicit `toString` defined to make the example more obvious. Like this:

```
val obj = object { override fun toString(): String = "Hello world" }
println("obj = $obj")
// obj = Hello world
```

File: docs/topics/strings.md [hunk 2]
```diff
@@ -132,6 +136,9 @@ fun main() {
     val s = "abc"
     println("$s.length is ${s.length}") 
     // abc.length is 3
+    val numbers = listOf(1,2,3,4,5)
+    println("Last two numbers: ${numbers.takeLast(2)}")
+    // Last two numbers: [4, 5]
```
Comment (nikitabobko) [inline docs/topics/strings.md:141]: I think this addition is redundant. We already made a point about `toString` in the previous example

**Review by nikitabobko — CHANGES_REQUESTED** [review]
Thanks for the elaboration, yes I agree that this part of the doc can be improved. `toString` is not mentioned in the "String templates" section at all, but it'd be good to mention it

> underlying mechanism can't be understood using Ctrl+Click somewhere

I like this benchmark

---

Overall the idea of the PR looks good to me except for the two comments that I left. 

@sarahhaggarty do you have any comments?

File: docs/topics/strings.md (hunk 1 above)
Comment (pichuzhkin) [reply docs/topics/strings.md:125]: Oh, really, that's very nice - I'm not so familiar with anonymous objects so I couldn't even think of such an example. Maybe we should keep my example and add yours also?

File: docs/topics/strings.md (hunk 2 above)
Comment (pichuzhkin) [reply docs/topics/strings.md:141]: I agree, that's redundant. Thanks!

File: docs/topics/strings.md (hunk 1 above)
Comment (nikitabobko) [reply docs/topics/strings.md]: > Maybe we should keep my example and add yours also?

What benefit does `listOf` example add?

Comment (pichuzhkin) [reply docs/topics/strings.md]: > What benefit does `listOf` example add?

I supposed that demonstration of all three behaviours (let's call it three levels of difficulty) will be most illustrative:
* Easy level - A simple int 
* Medium level -  A collection type 
* Hard level - An overridden toString() - thank you for this contribution

I won't insist on adding listOf() example if I don't convince you with this comment

**Review by sarahhaggarty — APPROVED** [review]

**Comment by sarahhaggarty** [comment]
Thanks for your contribution and insight @pichuzhkin! I added some changes on top of yours to match our style guide. Thanks again!

**Pull request merged**